jerry-core/ecma/builtin-objects/ecma-builtin-array.inc.h

```diff
@@ -19,6 +19,8 @@
 
 #include "ecma-builtin-helpers-macro-defines.inc.h"
 
+#ifndef CONFIG_DISABLE_ARRAY_BUILTIN
+
```

Comment:
You are oversimplifying here. The script is processing all files only once and does not start "grepping" other files all over the project directory structure to find out whether the current file is included elsewhere. And checking the first condition in the including file can be either a dangerous heuristic (in case some extra `#if`s sneak in the source file) or insufficient (we have at least one source where we have two guards in the source file (ecma-builtin-float64array-prototype.c)

```c
#ifndef CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN
#if CONFIG_ECMA_NUMBER_TYPE == CONFIG_ECMA_NUMBER_FLOAT64
```